feat: (back) 문제 리스트 관련 로직 (완료)

milda-regret opened the pull request:
## Summary
데이터베이스에서 문제 데이터를 추출해서 전달해주는 api를 구현중입니다.
## Related Issue
code-test generating system
## Description(optional)
POST api/problem : DB에 새 문제가 저장됩니다.
GET api/problem?difficulty={int}&num={int} : 난이도에 맞는 문제의 id를 개수 이하로 있는 만큼 반환합니다.
GET api/problem/[id] : id에 해당하는 문제의 정보를 반환합니다.

milda-regret committed:
feat: (back) initialize problem domain

add template files of problem to /server/src/problem/

milda-regret committed:
feat: (back) initialize problem domain

add template files of problem to /server/src/problem/

anzmin commented on server/src/problem/problem.controller.spec.ts:
```diff
@@ -0,0 +1,20 @@
+	it('should be defined', () => {
+		expect(controller).toBeDefined();
+	});
+});
```
*.spec과 같은 test 파일은 일단 지우셔도 됩니다.

anzmin commented on server/src/problem/dto/create-problem.dto.ts:
```diff
@@ -0,0 +1 @@
+export class CreateProblemDto {}
```
*.spec 파일들 처럼 실제로 쓰지 않는 파일은 지워주세요

milda-regret committed:
feat: (back) now we can add problem metadata to DB

milda-regret committed:
Merge branch 'problem' of github.com:skkuse/2023spring_41class_team3 into problem

milda-regret committed:
feat: (back) now you can create new problem metadata to DB

milda-regret committed:
feat: (back) now you can upload full problem includes description to DB

milda-regret committed:
feat: (back) add api for search problem with id

milda-regret committed:
refactor: (back) change func name findOneMeta to findOne

anzmin commented on server/src/problem/dto/update-problem.dto.ts:
```diff
@@ -0,0 +1,4 @@
+import { PartialType } from '@nestjs/mapped-types';
+import { CreateProblemDto } from './create-problem.dto';
+
+export class UpdateProblemDto extends PartialType(CreateProblemDto) {}
```
실제로 쓰지 않는 파일은 제거해주세요

anzmin commented on server/src/problem/entities/problem.entity.ts:
```diff
@@ -0,0 +1,27 @@
+export class ProblemEntity {
+	@Prop({ required: true, unique: true })
+	problemN: number;
+
```
주로 DB row의 id는 Id를 붙여서 명시합니다. problemId가 좋을 것 같아요

anzmin commented on server/src/problem/problem.controller.ts:
```diff
@@ -0,0 +1,38 @@
+	@Delete(':id')
+	remove(@Param('id') id: string) {
+		return this.problemService.remove(+id);
+	}
```
안 쓰는 메서드는 다 제거 해주세요

anzmin commented on server/src/problem/repository/problem.repository.ts:
```diff
@@ -0,0 +1,34 @@
+			const createdProblem = new this.problemModel(problem);
+			return await createdProblem.save();
+		} catch (error) {
+			throw new Error(error);
```
여기서 Error가 node의 Error를 쓴 것 같은데, 그 Error가 아니라 mongoose의 Error를 import해서 써야 mongoose exception filter에 catch 됩니다. `import { Error, Model } from 'mongoose';`로 추가해주세요

milda-regret replied:
수정했습니다!

milda-regret committed:
refactor: (back) delete unused methods and files

milda-regret committed:
fix: (back) change JS Error to Mongoose Error

change Error class in problem repository from JS Error to Mongoose Error

milda-regret committed:
feat: (back) add getCodeSet api for code-test problem set

milda-regret committed:
feat: (back) now findOne api return full problem data

now findOne api return full problem data not only metadata

milda-regret committed:
refactor: (back) delete more unused file

delete unused spec files

milda-regret committed:
refactor: (back) change method name for GET /api/problem

change method name for GET
/api/problem?difficulty=<difficulty>&num=<num> from getCodeSet to
getCodeSetIDs

milda-regret committed:
Merge branch 'dev' into problem
